### PR TITLE
[WIP] Travis simplifications

### PR DESCRIPTION
**Pull request by adidalal**
Since this repo doesn't have core code anymore, we can remove all the ruby/gem references

### CODE REVIEW

File: ci/travis/before_script.sh [hunk 1]
```diff
@@ -10,46 +10,13 @@
...
-# unset rvm hook functions
-run unset -f cd gem
-
-# print all travis-defined environment variables
-run 'env | sort'
-
-# print detailed macOS version info
-run sw_vers
-
-# capture system ruby and gem locations
```
Comment (adidalal) [inline ci/travis/before_script.sh]: Travis is still failing, and it's definitely due to this removal, any thoughts on a less convoluted way to do things?
@reitermarkus @jawshooah

File: .travis.yml [hunk 2]
```diff
@@ -22,10 +17,6 @@ branches:
   only:
     - master
 
-cache:
-  directories:
-    - $HOME/.gem
```
Comment (reitermarkus) [inline .travis.yml]: I *think* this is still used because `brew cask style` will install `rubocop-cask` in there.

**Review by reitermarkus — CHANGES_REQUESTED** [review]

File: .travis.yml (hunk 2 above)
Comment (reitermarkus) [inline .travis.yml]: Unless `cache: bundler` is the equivalent to this.

Comment (adidalal) [inline .travis.yml]: looks like `vendor/bundle` https://docs.travis-ci.com/user/caching/#Enabling-Bundler-caching

Comment (reitermarkus) [inline .travis.yml]: This won't work then, since we're not even using `bundler`.

File: ci/travis/before_script.sh [hunk 3]
```diff
@@ -10,46 +10,30 @@
 
 header 'Running before_script.sh...'
 
-# https://github.com/rvm/rvm/pull/3627
+# work around travis xcode bug
+# https://github.com/direnv/direnv/issues/210
```
Comment (reitermarkus) [inline ci/travis/before_script.sh]: https://github.com/travis-ci/travis-ci/issues/6307#issuecomment-233315824 might be a better link.

**Comment by adidalal** [comment]
Splitting this up into smaller PRs.

**Pull request closed without merging**